api/src/main/java/org/apache/iceberg/types/TypeUtil.java

```diff
@@ -42,6 +42,45 @@
   private TypeUtil() {
   }
 
+  /**
+   * Project extracts particular fields from a schema. Unlike
+   * {@link TypeUtil#select(Schema, Set)}, project will pick out only
+   * the fields enumerated, this means no sub fields will be selected
+   * unless they are explicitly requested.
```

Comment:
You might add "... unless they are explicitly requested. Structs that are explicitly projected are empty unless sub-fields are explicitly projected. Maps and lists cannot be explicitly selected in fieldIds."